---
title: Pathogen genomic surveillance
---
::: {.callout-tip}
#### Learning Objectives
At the end of this session you will be introduced to:

- what is genomic surveillance
- genomic surveillance techniques
- genomic sequencing and whole genome sequence
- key questions to answer in genomic surveillance.
:::

Genomics has been applied to tracking infectious diseases by sequencing the genomes of bacteria, parasites and viruses. According to WHO, genomic surveillance is the process of constantly monitoring pathogens and analyzing their genetic similarities and differences. From this perspective, WHO has prepared a 10-years global strategy for genomic surveillance of pathogens with pandemic and epidemic potential which you can read from [here](https://www.who.int/publications/i/item/9789240046979).

The underlying objective of the genomic surveillance is to provide information about pathogen evolution and insight into its genetic changes which might have impact on transmission or resistance to interventions.

![Overview of genomic surveillance. Source:[WHO](https://www.who.int/initiatives/genomic-surveillance-strategy)](images/genomic-surveillance.jpg){#fig-genomic-surveillance}

## Genomic sequencing for pathogen surveillance
One of the techniques used in genomic surveillance is **genomic sequencing**. In genomic sequencing we look into the DNA of pathogens to examine how they spread through communities, cause illnesses, and become resistant to medications and vaccines designed to stop them.

### Whole Genome Sequencing (WGS)

One of the techniques used in the pathogem genomic surveillance is **whole genome sequencing** in which we are interested to know the sequence or order of the DNA bases of the entire pathoge's genome. Today WGS has become a defacto standard approach in detecting and investigating outbreaks. For public health laboratories the main aim is to identify outbreak quickly by detecting the source of infection as well as providing information on antimicrbial resistance.

![Overview of the whole genome sequencing for detecting outbreaks. Source: [US CDC](https://www.cdc.gov/)](images/Genome-Sequencing.png){#fig-genome-sequencing}

Normally, the final output of the WGS are millions of DNA fragments technically known as sequence reads which are generated by sequencer of your choice such as Illumina, Oxford Nanopore or PacBio.

### Bioinformatics analysis

Once the WGS has been done the next stage is bioinformatics and further downstream analysis. This include the following:-

- Quality check and filtering of the sequence reads
- Genome assembly
- Checking the genome assembly quality
- Downstream analyses

All of the above analytical steps are discussed in the next sections.

## What are key genomic surveillance questions?

When an outbreak occurs the following are generally important public health questions that need to be adddressed during genomic surveillance. We use cholera as an example of AWD under investigation.

- Is there any Vibrio cholerae strain in the collected and sequenced samples?
- If the answer is yes, then is it a known pathogenic strain or unknown?
- If the answer is known, then does it known to be likely resistant to any antibiotics?
- If the answer is yes, then what kind of antibiotics resistant genes may contain and which antibiotics is predicted to be affected?

## Summary
In this section we have learned about the pathogen genomic surveillance and key approach used in genomic surveillance, which is whole genomic sequence (WGS). In the next section you will introuduced to the basics if the Unix command line.

::: {.callout-tip}
#### Key Points
- Pathogen genomic surveillance enables quick detection of the source of the outbreaks and provide insights on the transmission and antimicrobial resistance.
- Whole genome sequence (WGS) is one of the useful techniques to perform genomic surveillance.
- 
:::
