---
title: "Sub-regional training for national reference laboratories on
advanced laboratory diagnostics for AWD and genomic analysis for pathogen surveillance"
author: "Bajuna Salehe, Hugo Tavares"
date: today
number-sections: false
---

## Overview 

According to US [CDC](https://www.cdc.gov/), outbreaks of acute watery diarrhea (AWD) related diseases are likely to occur unless the surveillance systems are in place to rapidly detect associated pathogen(s) and an increase in reported cases. The surveillance system will also likely to trigger efforts to determine the source of transmission, ensure implementation of control measures in the affected area, and determine the microbial etiology.

These materials cover the genomic analysis for pathogen surveillance focusing on the bioinformatics analysis from raw data pre-processing to identification of typing and determination of antibiotics resistance for cholera as a case study, but could be applied for other pathogens causing infectious diseases, however, the wet laboratory training aspect of surveillance including raw genomic data generation from cholera cultured samples will not be covered in the materials. 

Cholera is an acute diarrhea disease which is caused by infection of the intestine with the pathogenic bacterium Vibrio cholerae serogroup O1 or O139. The cholera bacterium is usually found in water or in foods that have been contaminated by feces (poop) from a person infected with cholera bacteria or can be mostly likely occur in the aquatic environment naturally. Cholera is most likely to occur and spread in places with inadequate water treatment, poor sanitation, and inadequate hygiene. An estimated 1.3 to 4 million people around the world get cholera each year and 21,000 to 143,000 people die from it. Normally, people who are infected with this bacterium often show mild symptoms or no symptoms of the illness, however cholera can be very severe. Approximately 1 in 10 people who get sick with cholera will develop severe symptoms such as watery diarrhea, vomiting, and leg cramps. In these people, rapid loss of body fluids leads to dehydration and shock. Without proper treatment, death can occur very quickly. So having cholera surveillance system supported with bioinformatics capabilities is vital for public health.

We will start with the Introduction to Bioinformatics, which aims to introduce an audience to the basics bioinformatics techniques for genomic surveillance. The audience will be firstly introduced to work on the UNIX Command environment covering basics UNIX commands that are useful for bioinformatics analysis.

::: {.callout-tip}
### Learning Objectives

The overall objectives of the course are following:-

- Performing laboratory techniques used in extracting bacteria isolates from the suspected cholera patient.
- Performing barcoding with the ONT rapid barcoding kit.
- Performing quantification and adaptor ligations with the ONT data.
- Learning sampling library techniques including end-to-end library preparation.
- With case study, performing genomic approach (Whole Genome Sequencing or WGS) to identify pathogenic strain associated with Acute Watery Diarhea (AWD), that is Cholera in our context.
:::

### Target Audience

This course is primarily aimed at public health officials including doctors, lab workers and clinicians who work with waterbone or foodborne diseases in particular cholera with no prior experience in the bioinformatics tools, and would like to get started in using genomic and bioinformatics approaches for cholera or any other related pathogen surveillance.

### Prerequisites

- Basic understanding of computer and use of internet.
- Basic understanding of biology of bacteria (bacteriology).
- A working knowledge of the UNIX command line will be advantageous but not required as there will be a crash UNIX course before the training and significant numbers of sessions on UNIX command line during training.

<!-- Training Developer note: comment the following section out if you did not assign levels to your exercises -->
### Exercises

Exercises in these materials are labelled according to their level of difficulty:

| Level | Description |
| ----: | :---------- |
| {{< fa solid star >}} {{< fa regular star >}} {{< fa regular star >}} | Exercises in level 1 are simpler and designed to get you familiar with the concepts and syntax covered in the course. |
| {{< fa solid star >}} {{< fa solid star >}} {{< fa regular star >}} | Exercises in level 2 combine different concepts together and apply it to a given task. |
| {{< fa solid star >}} {{< fa solid star >}} {{< fa solid star >}} | Exercises in level 3 require going beyond the concepts and syntax introduced to solve new problems. |


## Authors
<!-- 
The listing below shows an example of how you can give more details about yourself.
These examples include icons with links to GitHub and Orcid. 
-->

About the authors:

- **Bajuna Salehe**
  <a href="https://github.com/bsalehe" target="_blank"><i class="fa-brands fa-github" style="color:#4078c0"></i></a>  
  _Affiliation_: Bioinformatics Training Facility, University of Cambridge  
  _Roles_: writing - original content; conceptualisation; coding
- **Hugo Tavares**
  <a href="https://orcid.org/0000-0001-9373-2726" target="_blank"><i class="fa-brands fa-orcid" style="color:#a6ce39"></i></a> 
  <a href="https://github.com/tavareshugo" target="_blank"><i class="fa-brands fa-github" style="color:#4078c0"></i></a>  
  _Affiliation_: Bioinformatics Training Facility, University of Cambridge  
  _Roles_: writing - review & editing; conceptualisation; coding


## Citation

<!-- We can do this at the end -->

Please cite these materials if:

- You adapted or used any of them in your own teaching.
- These materials were useful for your research work. For example, you can cite us in the methods section of your paper: "We carried our analyses based on the recommendations in _TODO_.".

You can cite these materials as:

> TODO

Or in BibTeX format:

```
@Misc{,
  author = {},
  title = {},
  month = {},
  year = {},
  url = {},
  doi = {}
}
```


## Acknowledgements

<!-- if there are no acknowledgements we can delete this section -->

- List any other sources of materials that were used.
- Or other people that may have advised during the material development (but are not authors).
