---
title: Case Study: Sudan cultured samples (Nanopore)
---

::: {.callout-tip}
#### Objectives

This section demonstrates a start-to-finish cholera genomic surveillance analysis of a cultured dataset from Sudan which was subsequently sequenced on a Nanopore platform. You can download the data from the link below:

    Switzerland Case Study - Full Data – this includes data for 65 samples, which gives a more realistic sample size, but can take several hours to run on a small computer.
    Switzerland Case Study - Small Version – this includes data for a subset of 10 samples, which is more suitable for training purposes (but the results will look slightly different from the ones shown here).

By the end of this section, you should be able to:

    Prepare all the files necessary to run the consensus pipeline.
    Run the viralrecon pipeline to generate FASTA consensus from raw FASTQ files.
    Assess and collect several quality metrics for the consensus sequences.
    Clean output files, in preparation for other downstream analysis.
    Assign sequences to lineages using Nextclade and/or Pangolin.
    Contextualise your sequences in other background data and cluster them based on phylogenetic analysis.
    Integrate the metadata and results to generate useful visualisations of your data.
    Report your analysis.


:::


## Section

Headings for material sections start at level 2. 

More guidelines for content available here: https://cambiotraining.github.io/quarto-course-template/materials/02-content_guidelines.html


## Summary

::: {.callout-tip}
#### Key Points

- Last section of the page is a bulleted summary of the key points
:::
